File: jpms-java/src/test/java/example/jpms_java/InsertTest.java
package example.jpms_java;

import example.jpms_java.domain.Age;
import example.jpms_java.domain.Salary;
import example.jpms_java.entity.Employee;
import example.jpms_java.entity.JobType;
import example.jpms_java.repository.EmployeeRepository;
import java.time.LocalDateTime;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;
import org.seasar.doma.jdbc.Config;

@ExtendWith(TestEnvironment.class)
public class InsertTest {

  private final EmployeeRepository repository;

  InsertTest(Config config) {
    repository = new EmployeeRepository(config);
  }

  @Test
  public void testInsert() {
    Employee employee = new Employee();
    employee.setName("test");
    employee.setAge(new Age(50));
    employee.setSalary(new Salary(300));
    employee.setJobType(JobType.PRESIDENT);
    repository.insert(employee);
  }

  @Test
  public void testInsertByNativeSql() {
    Employee employee = new Employee();
    employee.setId(100);
    employee.setName("test");
    employee.setAge(new Age(50));
    employee.setSalary(new Salary(300));
    employee.setJobType(JobType.PRESIDENT);
    employee.setInsertTimestamp(LocalDateTime.now());
    employee.setVersion(1);
    repository.insertByNativeSql(employee);
  }
}

File: dsl-style-kotlin/src/test/kotlin/example/dsl/style/kotlin/InsertTest.kt
package example.dsl.style.kotlin

import example.dsl.style.kotlin.domain.Age
import example.dsl.style.kotlin.domain.Salary
import example.dsl.style.kotlin.entity.Employee
import example.dsl.style.kotlin.entity.JobType
import example.dsl.style.kotlin.repository.EmployeeRepository
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.extension.ExtendWith
import org.seasar.doma.jdbc.Config
import java.time.LocalDateTime

@ExtendWith(TestEnvironment::class)
class InsertTest internal constructor(config: Config) {

    private val repository: EmployeeRepository = EmployeeRepository(config)

    @Test
    fun testInsert() {
        val employee = Employee()
        employee.name = "test"
        employee.age = Age(50)
        employee.salary = Salary(300)
        employee.jobType = JobType.PRESIDENT
        repository.insert(employee)
    }

    @Test
    fun testInsertByNativeSql() {
        val employee = Employee()
        employee.id = 100
        employee.name = "test"
        employee.age = Age(50)
        employee.salary = Salary(300)
        employee.jobType = JobType.PRESIDENT
        employee.insertTimestamp = LocalDateTime.now()
        employee.version = 1
        repository.insertByNativeSql(employee)
    }
}
